library/alloc/src/vec/mod.rs

```diff
@@ -2392,26 +2392,49 @@ impl<T: Clone, A: Allocator> Vec<T, A> {
         self.spec_extend(other.iter())
     }
 
-    /// Copies elements from `src` range to the end of the vector.
+    /// Appends elements specified by the `src` parameter to the end of the vector.
+    /// The `src` parameter is of type [`RangeBounds`], which means that
+    /// it can be used to specify a range of elements in the vector.
+    ///
+    /// [`RangeBounds`]: ../../std/ops/trait.RangeBounds.html
     ///
     /// # Panics
     ///
-    /// Panics if the starting point is greater than the end point or if
-    /// the end point is greater than the length of the vector.
+    /// Panics if the starting index is greater than the end index.
+    /// Panics if the end index is greater than the length of the vector.
     ///
     /// # Examples
     ///
+    /// With numbers:
+    ///
     /// ```
-    /// let mut vec = vec![0, 1, 2, 3, 4];
+    /// let mut numbers1 = vec![0, 1, 2, 3, 4];
+    /// numbers1.extend_from_within(2..);
+    /// assert_eq!(numbers1, [0, 1, 2, 3, 4, 2, 3, 4]);
+    ///
+    /// let mut numbers2 = vec![0, 1, 2, 3, 4];
+    /// numbers2.extend_from_within(..2);
+    /// assert_eq!(numbers2, [0, 1, 2, 3, 4, 0, 1]);
     ///
-    /// vec.extend_from_within(2..);
-    /// assert_eq!(vec, [0, 1, 2, 3, 4, 2, 3, 4]);
+    /// let mut numbers3 = vec![0, 1, 2, 3, 4];
+    /// numbers3.extend_from_within(1..3);
+    /// assert_eq!(numbers3, [0, 1, 2, 3, 4, 1, 2]);
```

Comment:
If the goal is to pervent confusion between the indices and values then this section can just be replaced, no need to keep it.

Reply:
done thx